hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java

```diff
@@ -1133,6 +1137,7 @@ public boolean swapTask(JobTask task, String replicaHostID, boolean kickOnComple
             return false;
         }
         Job job;
+        // Maybe
```

Comment:
Strongly believe that this lock should be kept.

If we somehow tried to swap in two threads at the same time (swaps can be executed via an HTTP call, I believe) then modifications to the replicas and the hostUUID could interleave in a very harmful way -- for example, setting the replica and the host to the same UUID and "losing" a valid replica elsewhere.